File: LeetCode-JavaScript/53-最大子序和.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let sum = 0
    let max_sum = nums[0]
    for (let i = 0;i < nums.length;i++) {
        if (sum < 0) {
            sum = nums[i]
        } else {
            sum += nums[i]
        }
        if (sum > max_sum) {
            max_sum = sum
        }
    }
    return max_sum
};

File: LeetCode-Go/53-最大子序和.go
package main

import "fmt"

func main() {
 fmt.Println(maxSubArray([]int{-2,1,-3,4,-1,2,1,-5,4}))
}
func maxSubArray(nums []int) int {
	sum := 0
	max_sum := nums[0]
	for i:=0;i < len(nums); i++ {
		if sum < 0{
			sum = nums[i]
		} else {
			sum += nums[i]
		}
		if sum > max_sum{
			max_sum = sum
		}
	}
	return max_sum
}

